cpu/stm32_common/stm32_mem_lengths.mk

```diff
@@ -0,0 +1,304 @@
+STM32_TYPE      = $(word 1, $(STM32_INFO))
+STM32_FAMILY    = $(word 2, $(STM32_INFO))
+STM32_MODEL     = $(word 3, $(STM32_INFO))
+STM32_MODEL2    = $(word 4, $(STM32_INFO))
+STM32_MODEL3    = $(word 5, $(STM32_INFO))
+STM32_PINCOUNT  = $(word 6, $(STM32_INFO))
+STM32_ROMSIZE   = $(word 7, $(STM32_INFO))
+
+ifeq ($(STM32_TYPE), F)
+  ifeq ($(STM32_FAMILY), 0)
+    ifeq ($(STM32_MODEL2), 3)
+      ifneq (, $(filter $(STM32_MODEL3), 0))
+        ifneq (, $(filter $(STM32_ROMSIZE), 4 6))
+          RAM_LEN = 4K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 8))
+          RAM_LEN = 8K
+        else ifneq (, $(filter $(STM32_ROMSIZE), C))
+          RAM_LEN = 32K
+        endif
+      else ifneq (, $(filter $(STM32_MODEL3), 1 8))
+        RAM_LEN = 4K
+      endif
+    else ifneq (, $(filter $(STM32_MODEL2), 4))
+      RAM_LEN = 6K
+    else ifneq (, $(filter $(STM32_MODEL2), 5))
+      RAM_LEN = 8K
+    else ifneq (, $(filter $(STM32_MODEL2), 7))
+      ifneq (, $(filter $(STM32_MODEL3), 0))
+        ifneq (, $(filter $(STM32_ROMSIZE), 6))
+          RAM_LEN = 6K
+        else ifneq (, $(filter $(STM32_ROMSIZE), B))
+          RAM_LEN = 16K
+        endif
+      else ifneq (, $(filter $(STM32_MODEL3), 1 2 8))
+        RAM_LEN = 16K
+      endif
+    else ifneq (, $(filter $(STM32_MODEL2), 9))
+      RAM_LEN = 32K
+    endif
+  else ifeq ($(STM32_FAMILY),1)
+    ifeq ($(STM32_MODEL2), 0)
+      ifeq ($(STM32_MODEL3), 0)
+        ifneq (, $(filter $(STM32_ROMSIZE), 4 6))
+          RAM_LEN = 4K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 8 B))
+          RAM_LEN = 8K
+        else ifneq (, $(filter $(STM32_ROMSIZE), C))
+          RAM_LEN = 24K
+        else ifneq (, $(filter $(STM32_ROMSIZE), E D))
+          RAM_LEN = 32K
+        endif
+      else ifneq (, $(filter $(STM32_MODEL3), 1 2))
+        ifneq (, $(filter $(STM32_ROMSIZE), 4))
+          RAM_LEN = 4K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 6))
+          RAM_LEN = 6K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 8))
+          RAM_LEN = 10K
+        else ifneq (, $(filter $(STM32_ROMSIZE), B))
+          RAM_LEN = 16K
+        else ifneq (, $(filter $(STM32_ROMSIZE), C))
+          RAM_LEN = 32K
+        else ifneq (, $(filter $(STM32_ROMSIZE), D E))
+          RAM_LEN = 48K
+        else ifneq (, $(filter $(STM32_ROMSIZE), F G))
+          RAM_LEN = 80K
+        endif
+      else ifeq ($(STM32_MODEL3), 3)
+        ifneq (, $(filter $(STM32_ROMSIZE), 4))
+          RAM_LEN = 6K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 6))
+          RAM_LEN = 10K
+        else ifneq (, $(filter $(STM32_ROMSIZE), 8 B))
+          RAM_LEN = 20K
+        else ifneq (, $(filter $(STM32_ROMSIZE), C D E))
+          RAM_LEN = 64K
+        else ifneq (, $(filter $(STM32_ROMSIZE), F G))
+          RAM_LEN = 96K
+        endif
+      endif
+    else ifneq (, $(filter $(STM32_MODEL), 105 107))
+        RAM_LEN = 64K
+    endif
+  else ifeq ($(STM32_FAMILY),2)
```

Comment:
it seems I was wrong for F2 as well, sorry.
F2x5xB -> 64K
F2x5xC -> 96K
F2x5x[E F G] -> 128K
F2x7xx -> 128K

It seems a bit weird for f2x7 but that's what the stm32 claims (http://www.st.com/content/st_com/en/products/microcontrollers/stm32-32-bit-arm-cortex-mcus/stm32-high-performance-mcus/stm32f2-series/stm32f2x7.html?querycriteria=productId=LN9)

Reply:
Oh, I didn't check for the F2. I'll modify.

Reply:
Done.